server/app/models/ProductionModel.ts

```diff
@@ -0,0 +1,51 @@
+import { Document, Schema } from 'mongoose';
+
+export interface IProduction extends Document {
+  type: string;
+  componentDetail: {
+    name: string;
+    description: string;
+    quality: string;
+    color: string;
+    parts: Record<string, string>;
+    type: string;
+    finish: string;
+    grade: string;
+    weightAmount: number;
+    weightType: string;
+    detail: string;
+  };
+  status: string;
+  quantity: string;
+  startDate: Date;
+  endDate: Date;
+  assemblyMachine: string;
+  note: string;
+}
+
+export const ProductionSchema: Schema = new Schema({
+  type: { type: Schema.Types.String, required: true, enum: ['Bike', 'Part'] },
+  componentDetail: {
+    name: { type: Schema.Types.String, required: true },
+    description: { type: Schema.Types.String, required: false },
+    quality: { type: Schema.Types.String, required: false },
+    color: { type: Schema.Types.String, required: false },
+    parts: { type: Schema.Types.Mixed, required: false },
+    type: { type: Schema.Types.String, required: false },
+    finish: { type: Schema.Types.String, required: false },
+    grade: { type: Schema.Types.String, required: false },
+    weightAmount: { type: Schema.Types.Number, required: false },
+    weightType: { type: Schema.Types.String, required: false, enum: ['kg', 'lb'] },
+    detail: { type: Schema.Types.String, required: false },
```

Comment:
this feels wrong, youre basically recreating a bike in a production here

Reply:
Right, this was actually a suggestion I gave. At the time I couldn't think of another way for the frontend to tell the backend what kind of bike and parts we'd like to put in production. I felt like this was a quick and dirty way to do it.

Maybe we can use a reference Mongo doc ID instead?

Reply:
ya the request body can contain all of this but I think we should just store the mongo ID in the production collection

Reply:
or we completely remove the Bike code we have on the server

Reply:
Oh, yeah you're 100% right. The request body needs this info but not the MongoDB document